src/libAtomVM/nifs.c

```diff
@@ -5842,6 +5851,50 @@ static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
     return result;
 }
 
+
+static term nif_file_get_cwd_0(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+    UNUSED(argv)
+
+    char cwd[PATH_MAX];
+    if (IS_NULL_PTR(getcwd(cwd, PATH_MAX))) {
+        if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2)) != MEMORY_GC_OK)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+        term result_tuple = term_alloc_tuple(2, &ctx->heap);
+        term reason = UNDEFINED_ATOM;
+        switch (errno) {
```

Comment:
POSIX error code are already standard in OTP, so we don't have to invent new codes:
https://www.erlang.org/doc/apps/kernel/file.html#t:posix/0
Also we have a functions that handles them, that is `posix_errno_to_term`.